skbio/sequence/_iupac_sequence.py

```diff
@@ -164,23 +165,72 @@ def _motifs(self):
 
     @overrides(Sequence)
     def __init__(self, sequence, metadata=None, positional_metadata=None,
-                 validate=True, case_insensitive=False):
+                 validate=True, lowercase=False):
         super(IUPACSequence, self).__init__(
             sequence, metadata, positional_metadata)
 
-        if case_insensitive:
+        # Note we are checking identity against the False object here and not
+        # just falsiness
+        if lowercase is not False:
+            self._lowercase_mask = self._bytes > self._ascii_lowercase_boundary
+            if not isinstance(lowercase, bool):
+                self.positional_metadata[lowercase] = self._lowercase_mask
             self._convert_to_uppercase()
 
         if validate:
             self._validate()
 
     def _convert_to_uppercase(self):
-        lowercase = self._bytes > self._ascii_lowercase_boundary
+        lowercase = self._lowercase_mask
         if np.any(lowercase):
             with self._byte_ownership():
-                # ASCII is built such that the difference between uppercase and
```

Comment:
This comment was useful, can you add it back either here or on `_ascii_invert_case_bit_offset`?